types/block.go

```diff
@@ -22,6 +22,8 @@ func BlockGasLimit(ctx sdk.Context) uint64 {
 	maxGas := cp.Block.MaxGas
 	if maxGas > 0 {
 		return uint64(maxGas)
+	} else if maxGas == -1 {
```

Comment:
the issue here is that EVMs should never use unlimited block gas. You can define a high amount manually in the consensus params